File: 1-easy/00066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        offset = 1
        digits_len = len(digits)

        for i in range(digits_len - 1, -1, -1):
            digits[i] += offset
            offset = 0
            if digits[i] > 9:
                digits[i] -= 10
                offset = 1
            else:
                break

        if offset == 1:
            digits.append(0)
            digits[0] = 1

        return digits

File: 1-easy/00066-plus-one/solution.rs
impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {

        let mut offset = 1;
        let digits_len = digits.len();

        for i in (0..digits_len).rev() {
            digits[i] += offset;
            offset = 0;
            if digits[i] > 9 {
                digits[i] -= 10;
                offset = 1;
            } else {
                break;
            }
        }

        if offset == 1 {
            digits.push(0);
            digits[0] = 1;
        }

        digits
    }
}
